cmake/BLTMacros.cmake

```diff
@@ -663,3 +663,56 @@ macro(blt_append_custom_compiler_flag)
 
 endmacro(blt_append_custom_compiler_flag)
 
+
+
+
+##------------------------------------------------------------------------------
+## blt_find_libraries( <VAR>
+##                     NAMES [libname1 [libname2 ...]]
+##                     PATHS [path1 [path2 ...]] )
+##
+## This command is used to find a list of libraries. A cache entry named by <VAR> 
+## is created to store the result of this command. If the libraries are found the
+## results are stored in VAR.
+##
+## NAMES lists the names of the libraries to search for.
+##
+## PATH lists the paths in which to search for NAMES.
+##
+##
+##------------------------------------------------------------------------------
+macro(blt_find_libraries VAR)
+
+    set(multiValueArgs NAMES
+                       PATHS )
+
+    ## parse the arguments
+    cmake_parse_arguments(arg
+        "${options}" "${singleValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+    if ( NOT DEFINED arg_NAMES )
+        message(FATAL_ERROR "blt_find_libraries requires that NAMES specify the library names you are searching for")
+    endif()
+
+    if ( NOT DEFINED arg_PATHS )
+        message(FATAL_ERROR "blt_find_libraries requires that PATHS specify the paths to search for NAMES")
+    endif()
+
+    foreach( lib ${arg_NAMES} )
+        unset( temp CACHE )
+        find_library( temp NAMES ${lib}
```

Comment:
It would be nice if this macro also handled foo and libfoo since that is used so often.

Reply:
so this I assume the best way to do this is to check to see if there is a "lib" in front of "foo" and strip it out prior to the call do camke::find_library()? Is it safe to just strip the characters from the name? i.e. is there something that might be called "liblibfoo" and it is entered as "libfoo" then the lib will get stripped off and it won't be found.

Reply:
Cyrus claims it does the correct thing already.  This might need to be tested.  I will do so now

I was more saying that it should add both the base name and the lib+base name to this find_library call.  Having extra names that don't match wont cause problems..

 For example:

set(_prependedlib "lib${lib}")
find_library( temp NAMES ${lib} ${_prependedlib}

Reply:
I apparently don't know what I am talking about. I could have sworn you needed both on the function.  But I just tested and if you add "foo libfoo" it works cuz it finds "foo" but if you do just "libfoo" it fails to find it.  So this is a non issue apparently.

Reply:
OK. I am going to leave it as is then.